src/demos/findex/findex.ts

```diff
@@ -0,0 +1,185 @@
+import { hexDecode, hexEncode, sanitizeString } from "../../utils/utils";
+
+export class FindexDemo {
+  private _db: DB;
+  private _policy: Policy;
+  private _masterKeysCoverCrypt: AbeMasterKey;
+
+  constructor(db: DB, policyAxes: PolicyAxis[], maxAttributeCreations: number) {
+    this._db = db;
+    this._policy = new Policy(policyAxes, maxAttributeCreations)
+    this._masterKeysCoverCrypt = generateMasterKeys(this._policy);
+  }
+
+  public get db(): DB {
+    return this._db;
+  }
+  public get policy(): Policy {
+    return this._policy;
+  }
+  public get masterKeysCoverCrypt(): AbeMasterKey {
+    return this._masterKeysCoverCrypt;
+  }
+
+  async encryptUsers(metadataUid: Uint8Array) {
+    const policyBytes = this._policy.toJsonEncoded();
+
+    const users = await this._db.getUsers();
+    for (const user of users) {
+      const encryptedBasic = coverCryptEncrypt(
+        policyBytes,
+        this._masterKeysCoverCrypt.publicKey,
+        metadataUid,
+        [`department::marketing`, `country::${user.country}`],
+        JSON.stringify({
+          firstName: user.firstName,
+          lastName: user.lastName,
+          country: user.country,
+          region: user.region
+        }))
+
+      const encryptedHr = coverCryptEncrypt(
+        policyBytes,
+        this._masterKeysCoverCrypt.publicKey,
+        metadataUid,
+        [`department::HR`, `country::${user.country}`], JSON.stringify({
+          email: user.email,
+          phone: user.phone,
+          employeeNumber: user.employeeNumber
+        })
+      )
+
+      const encryptedSecurity = coverCryptEncrypt(
+        policyBytes,
+        this._masterKeysCoverCrypt.publicKey,
+        metadataUid,
+        [`department::security`, `country::${user.country}`], JSON.stringify({
+          security: user.security
+        }))
+
+      const upsertedEncElement = await this._db.upsertEncryptedUser({
+        enc_basic: hexEncode(encryptedBasic),
+        enc_hr: hexEncode(encryptedHr),
+        enc_security: hexEncode(encryptedSecurity)
+      });
+
+      await this._db.upsertUserEncUidById(user.id, { enc_uid: upsertedEncElement[0].uid });
+    };
+
+  }
+
+  /**
+   * Reset all indexes and upsert new ones
+   *
+   * @param location location string naming the key of location to index
+   */
+  async resetAndUpsert(location: string) {
+    await this._db.deleteAllChainTableEntries();
+    await this._db.deleteAllEntryTableEntries();
+    const users: User[] = await this._db.getUsers();
+    const sanitizedElements = users.map((user) => {
+      let key: keyof typeof user;
+      for (key in user) {
+        if (user[key]) {
+          user[key] = sanitizeString(user[key])
+        }
+      }
+      return user;
+    })
+    const locationAndWords: { [key: string]: string[]; } = {};
+    sanitizedElements.map((user) => {
+      let userId = user.id;
+      if (location === "enc_uid") {
+        userId = user.enc_uid;
+      }
```

Comment:
This is not useful if you set the type of `location` as `keyof User`, or even `location: 'id' | 'enc_uid' = 'id'` if you want it only to be a choice between 'id' and 'enc_uid'